enhancements/installer/packet-ipi.md

```diff
@@ -0,0 +1,197 @@
+models of compute, memory, GPU, or storage classes of devices, hardware and IP
+reservations, virtual networks, and fast direct-attached storage.
+
+Currently, users who want to [deploy OpenShift on
+Packet](https://www.packet.com/solutions/openshift/) must follow the [OpenShift
+via Terraform on
+Packet][https://github.com/RedHatSI/openshift-packet-deploy/blob/master/terraform/README.md]
+UPI instructions. This process is not streamlined and can not be integrated into
+a simple experience like the [Try OpenShift 4](https://www.openshift.com/try)
+workflow.
+
+### Goals
+
+The main goal of the Packet IPI is to provide users with an easier path to
+running Red Hat OpenShift 4 on a horizontally scalable bare metal architecture
+in data centers.
+
+As a first step, we would add the Packet IPI to the installer codebase. With accompanying documentation, this makes a the Packet option available through the CLI. This would enable users to deploy OpenShift with IPI in a way
+that is very similar to UPI but simplifies the process.
+
+Following other IPI installers, this first step would include
+
+- Making Packet IPI documentation available here:
+  <https://github.com/openshift/installer/blob/master/docs/user/packet/install_ipi.md>
+- Adapt Packet's sig-lifecyclce ClusterAPI v1alpha2 for use as an OpenShift's ClusterAPI v1beta1 Machine driver
+- Prepare the Terraform code and Packet types necessary for an IPI installer
+- Making a CI job executing the provisioning scripts to test the IPI installer
+
+### Non-Goals
+
+It is outside the scope of this enhancement to provide explanations about the
+installation of infrastructure elements that are considered as required and
+owned by the user (e.g. DNS, Load Balancer).
+
+Additional enhancement requests will be created to determine the best approach for fulfilling each need.
+
+## Proposal
+
+- Define and implement Packet types in the openshift/installer
+  - High level variables should include:
+    - API Key
+    - Project ID
+  - Boostrap node variables include:
+    - Device plan (defaulting to minimum required)
+    - Facility
+    - Virtual Networks
+  - Control plane variables rely on boostrap variables and add:
+    - Device plan (defaulting to provide a good experience)
+  - Additional clusters creation should include
+    - Device plan (defaulting to provide a good experience)
+    - Facility
+    - Virtual Networks
+    - Future: IP Reservations, Hardware Reservations, Spot Market Requests
+- Write documentation to help users use and understand the Packet installer, to include:
+  - Packet usage basics (accounts and API keys)
+  - Packet resource requirements and options
+  - OpenShift Installer options
+  - Non-Packet components that build on the experience (DNS configuration)
+- Setup the CI job to have a running test suite
+
+### Implementation Details/Notes/Constraints
+
+- RHCOS must be made available and maintained.
+  Images available for provisioning may be custom or official.
```

Comment:
Packet's official image and custom image mechanisms are both based on filesystem tarballs rather than disk images, and RHCOS is delivered as a disk image.

If we want to get an official RHCOS image into Packet, we should work with the Packet folks to set up a custom deployment flow using [coreos-installer](https://github.com/coreos/coreos-installer/).  CoreOS Container Linux used a similar arrangement.

Otherwise, we can use Packet's custom iPXE support to implement the same thing ourselves.  We'd just boot the RHCOS live PXE image and perform an ordinary bare-metal install, telling the installer to override the platform ID to `packet`.

Reply:
I've captured most of this in the latest push. Thanks for the input and context, @bgilbert!